mycore-restapi/src/main/java/org/mycore/restapi/MCRSessionFilter.java

```diff
@@ -136,14 +136,31 @@ public void filter(ContainerRequestContext requestContext) {
         currentSession.setCurrentIP(MCRFrontendUtil.getRemoteAddr(httpServletRequest));
         MCRTransactionHelper.beginTransaction();
         //3 cases for authentication
-        Optional<MCRUserInformation> userInformation = Optional.empty();
         String authorization = requestContext.getHeaderString(HttpHeaders.AUTHORIZATION);
         //1. no authentication
         if (authorization == null) {
             LOGGER.debug("No 'Authorization' header");
             return;
         }
         //2. Basic Authentification
+        Optional<MCRUserInformation> userInformation
+            = getMcrUserInformation(requestContext, authorization, currentSession);
+
+        if (userInformation.isEmpty()) {
+            LOGGER.warn(() -> "Unsupported " + HttpHeaders.AUTHORIZATION + " header: " + authorization);
+        }
+
+        userInformation
+            .ifPresent(ui -> {
+                currentSession.setUserInformation(ui);
+                requestContext.setSecurityContext(new MCRRestSecurityContext(ui, isSecure));
+            });
+        LOGGER.info("user detected: " + currentSession.getUserInformation().getUserID());
+    }
```

Comment:
I would refactor the methods for getting the user information based on _Basic_ and _Bearer_ into separate methods. In general, the method currently refers to `HttpHeaders.AUTHORIZATION`, so I would call the respective method directly and not nest it.

```
MCRUserInformation userInformation = null;

if (authorization.startsWith(BASIC_AUTH_PREFIX)) {
    userInformation = extractUserFromBasicAuth(requestContext, authorization, MCRSession currentSession);
} else if (authorization.startsWith(BEARER_AUTH_PREFIX)) {
    userInformation = extractUserFromBearerAuth(requestContext, authorization, MCRSession currentSession);
} else {
    LOGGER.warn(() -> "Unsupported " + HttpHeaders.AUTHORIZATION + " header: " + authorization);
    return;
}

currentSession.setUserInformation(userInformation);
requestContext.setSecurityContext(new MCRRestSecurityContext(userInformation, isSecure));
LOGGER.info("user detected: " + currentSession.getUserInformation().getUserID());
```